SPECS/glibc/glibc.spec

```
@@ -251,19 +251,26 @@ ls -1 %{buildroot}%{_libdir}/*.a | grep -v -e "$static_libs_in_devel_pattern" |
 %check
 cd %{_builddir}/glibc-build
 
-# Should have the following results:
-# Summary of test results:
+# Results have varied based on the environment the tests are being built
+# Summary of test results in local VM:
 #      3 FAIL : nptl/tst-cancel1, io/tst-lchmod, nptl/tst-mutex10
 #   5040 PASS
 #    152 UNSUPPORTED
 #     12 XFAIL
 #      8 XPASS
+# Summary of test results in pipeline (this has shown varying results):
+#       7 FAIL
+#    5110 PASS
+#      79 UNSUPPORTED
+#      12 XFAIL
+#       8 XPASS
 make %{?_smp_mflags} check ||:
 n=0
+# expected failures in local VM
 grep "^FAIL: nptl/tst-cancel1" tests.sum >/dev/null && n=$((n+1)) ||:
 grep "^FAIL: io/tst-lchmod" tests.sum >/dev/null && n=$((n+1)) ||:
 grep "^FAIL: nptl/tst-mutex10" tests.sum >/dev/null && n=$((n+1)) ||:
-[ `grep ^FAIL tests.sum | wc -l` -ne $n ] && exit 1 ||:
+[ `grep ^FAIL tests.sum | wc -l` -eq $n ] ||:
```

The `||:` part will cause this to always return with an exist code of 0. We want the tests to fail, if the number of known failing tests don't match:
```suggestion
[ `grep ^FAIL tests.sum | wc -l` -eq $n ]
```